packages/gatsby-theme-docs/src/overrides/README.md

```diff
@@ -1,7 +1,57 @@
-Overrides files using theme shadowing, in order to inject specific functionalities to parts of the theme.
+Overrides files using [Gatsby theme shadowing](https://www.gatsbyjs.org/docs/themes/shadowing/), to inject functionalities to specific parts of the theme.
+
+## Usage
+
+To use this feature, the website should mimic the folder structure of the Gatsby theme and list the files in the `overrides` folder that should be overridden.
+
+```
+src
+└── @commercetools-docs
+    └── gatsby-theme-docs
+        └── overrides
+            ├── markdown-components.js
+            ├── page-header-side.js
+            └── use-additional-site-data.js
+```
+
+> Note that Gatsby shadowing works with any file in the theme. However, we want to provide an explicit extension point that consumers of the theme should use, instead of relying on internal file system paths that can potentially change any time.
```

Comment:
```suggestion
> Note that Gatsby shadowing works with any file in the theme. However, the theme provides an explicit stable extension point that is the `overrides` folder. Users of the theme should use this extension point only instead of relying on theme-internal file system paths that can potentially change any time.
```